test/pydmap_tests/test_dmapread.py

```diff
@@ -0,0 +1,325 @@
+        self.assertIsInstance(dm_records[5]['mppul'].value, int)
+        self.assertIsInstance(dm_records[6]['tnoise'].value, np.ndarray)
+        self.assertEqual(dm_records[7]['channel'].value, 0)
+        self.assertEqual(dm_records[10]['data'].dimension, 1)
+
+    def test_read_rawacf(self):
+        """
+        Test reading records from rawacf.
+
+        Checks:
+            - returns correct data structures
+            - returns excpected values
+        """
+        file_path = rawacf_file
+        dm = pydarn.DmapRead(file_path)
+        dm_records = dm.read_records()
+        self.assertIsInstance(dm_records, collections.deque)
+        self.assertIsInstance(dm_records[0], collections.OrderedDict)
+        self.assertIsInstance(dm_records[4]['channel'], pydarn.DmapScalar)
+        self.assertIsInstance(dm_records[1]['ptab'], pydarn.DmapArray)
+        self.assertIsInstance(dm_records[7]['channel'].value, int)
+        self.assertIsInstance(dm_records[2]['xcfd'].value, np.ndarray)
+        self.assertEqual(dm_records[0]['xcfd'].dimension, 3)
+
+    def test_read_fitacf(self):
+        """
+        Test reading records from fitacf.
+
+        Checks:
+            - returns correct data structures
+            - returns excpected values
+        """
+        file_path = fitacf_file
+        dm = pydarn.DmapRead(file_path)
+        dm_records = dm.read_records()
+        self.assertIsInstance(dm_records, collections.deque)
+        self.assertIsInstance(dm_records[0], collections.OrderedDict)
+        self.assertIsInstance(dm_records[2]['channel'], pydarn.DmapScalar)
+        self.assertIsInstance(dm_records[5]['ptab'], pydarn.DmapArray)
+        self.assertIsInstance(dm_records[9]['channel'].value, int)
+        self.assertIsInstance(dm_records[4]['ptab'].value, np.ndarray)
+        self.assertEqual(dm_records[0]['scan'].value, 1)
+        self.assertEqual(dm_records[7]['ltab'].dimension, 2)
+
+    def test_read_map(self):
+        """
+        Test reading records from map file.
+
+        Checks:
+            - returns correct data structures
+            - returns excpected values
+        """
+        file_path = map_file
+        dm = pydarn.DmapRead(file_path)
+        dm_records = dm.read_records()
+        self.assertIsInstance(dm_records, collections.deque)
+        self.assertIsInstance(dm_records[0], collections.OrderedDict)
+        self.assertIsInstance(dm_records[2]['IMF.flag'], pydarn.pydmap.datastructures.DmapScalar)
+        self.assertIsInstance(dm_records[3]['stid'], pydarn.DmapArray)
+        self.assertIsInstance(dm_records[8]['IMF.flag'].value, int)
+        self.assertIsInstance(dm_records[10]['stid'].value, np.ndarray)
+        self.assertEqual(dm_records[3]['stid'].dimension, 1)
+        self.assertEqual(dm_records[0]['stid'].shape[0], 14)  # this will be file dependent... future working test project.
+
+    # TODO: Again dependent on the file used :/
```

Comment:
Force a file with some corrupt records for testing. I would just read in a file, scramble the buffer a bit and then rewrite the binary.

Reply:
If you read further into the file you see a FIXME which is indicating I could not get that rescambling stream unit test to work. Also it took 30 minutes to run  :/ 

Also, this PR is not focused on the test suite just yet because I need to figure out a way for the files to be reachable for anyone that wants to run the test suite. I did this merely to ensure that the code worked before pushing it out.